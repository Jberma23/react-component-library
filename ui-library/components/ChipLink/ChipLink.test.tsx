import { render, screen } from "@testing-library/react";
import { ChipLink } from ".";

// Since this is just a sample component, the tests below are not intended to be comprehensive
describe("components/Chip/Chip.tsx", () => {
  it("renders with defaults", () => {
    const { asFragment } = render(<ChipLink href="/foo/bar">Chip</ChipLink>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("appends additional classnames", () => {
    render(
      <ChipLink className={"test-class"} href="/foo/bar">
        Chip
      </ChipLink>
    );
    const ChipClass = screen.getByText("Chip")?.getAttribute("class");
    expect(ChipClass).toContain("test-class");
  });
});
