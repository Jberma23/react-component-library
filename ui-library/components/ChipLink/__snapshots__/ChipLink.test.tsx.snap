// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Chip/Chip.tsx renders with defaults 1`] = `
<DocumentFragment>
  <a
    class="rounded-3 font-semibold outline-none focus-visible:shadow-chip transition-colors duration-default"
    href="/foo/bar"
  >
    Chip
  </a>
</DocumentFragment>
`;
